src/properties/ecc.c

```diff
@@ -0,0 +1,349 @@
+}
+
+static igraph_error_t igraph_i_ecc4(
+        const igraph_t *graph, igraph_vector_t *res,
+        const igraph_es_t eids, igraph_bool_t offset, igraph_bool_t normalize) {
+
+    igraph_integer_t no_of_nodes = igraph_vcount(graph);
+    igraph_vector_int_t degree;
+    igraph_adjlist_t al;
+    igraph_eit_t eit;
+    igraph_real_t c = offset ? 1.0 : 0.0;
+
+    IGRAPH_CHECK(igraph_adjlist_init(graph, &al, IGRAPH_ALL, IGRAPH_NO_LOOPS, IGRAPH_NO_MULTIPLE));
+    IGRAPH_FINALLY(igraph_adjlist_destroy, &al);
+
+    for (igraph_integer_t i=0; i < no_of_nodes; i++) {
+        igraph_vector_int_sort(igraph_adjlist_get(&al, i));
+    }
+
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&degree, no_of_nodes);
+    IGRAPH_CHECK(igraph_degree(graph, &degree, igraph_vss_all(), IGRAPH_ALL, IGRAPH_LOOPS));
+
+    IGRAPH_CHECK(igraph_eit_create(graph, eids, &eit));
+    IGRAPH_FINALLY(igraph_eit_destroy, &eit);
+
+    IGRAPH_CHECK(igraph_vector_resize(res, IGRAPH_EIT_SIZE(eit)));
+
+    for (igraph_integer_t i=0;
+         ! IGRAPH_EIT_END(eit);
+         IGRAPH_EIT_NEXT(eit), i++) {
+
+        igraph_integer_t edge = IGRAPH_EIT_GET(eit);
+        igraph_integer_t v1 = IGRAPH_FROM(graph, edge), v2 = IGRAPH_TO(graph, edge);
+
+        igraph_real_t z; /* number of 4-cycles the edge participates in */
+        igraph_real_t s; /* max number of 4-cycles the edge could be part of */
+
+        IGRAPH_ALLOW_INTERRUPTION();
+
+        if (v1 == v2) {
+            z = 0.0;
+            s = 0.0;
+        } else {
+            /* ensure that v1 is the vertex with the smaller degree */
+            if (VECTOR(degree)[v1] > VECTOR(degree)[v2]) {
+                igraph_integer_t tmp = v1;
+                v1 = v2;
+                v2 = tmp;
+            }
+
+            z = 0.0;
+            const igraph_vector_int_t *a1 = igraph_adjlist_get(&al, v1);
+            const igraph_integer_t n = igraph_vector_int_size(a1);
+            for (igraph_integer_t j=0; j < n; j++) {
+                igraph_integer_t v3 = VECTOR(*a1)[j];
+
+                /* It is not possible that v3 == v1 because self-loops have been removed from the adjlist. */
+
+                if (v3 == v2) continue;
+
+                const igraph_vector_int_t *a2 = igraph_adjlist_get(&al, v2), *a3 = igraph_adjlist_get(&al, v3);
+
+                z += vector_int_intersection_size_sorted(a2, a3) - 1.0;
+            }
+
+            igraph_integer_t d1 = VECTOR(degree)[v1], d2 = VECTOR(degree)[v2];
+            s = (d1 - 1.0) * (d2 - 1.0);
+        }
+
+        VECTOR(*res)[i] = z + c;
+        if (normalize) VECTOR(*res)[i] /= s;
+    }
+
+    igraph_eit_destroy(&eit);
+    igraph_vector_int_destroy(&degree);
+    igraph_adjlist_destroy(&al);
+    IGRAPH_FINALLY_CLEAN(3);
+
+    return IGRAPH_SUCCESS;
+}
+
+
+/**
+ * \function igraph_ecc
+ * \brief Edge clustering coefficient of some edges.
+ *
+ * \experimental
+ *
+ * The edge clustering coefficient <code>C^(k)_ij</code> of an edge $(i, j)$
```

Comment:
I think that the `$(i, j)$` LaTeX syntax won't work in the documentation, right?